packages/fields/src/fields/featured-image/featured-image-view.tsx

```diff
@@ -0,0 +1,176 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useCallback } from '@wordpress/element';
+import type { Attachment } from '@wordpress/core-data';
+import { store as coreStore, useEntityRecord } from '@wordpress/core-data';
+// @ts-ignore
+import { privateApis as routerPrivateApis } from '@wordpress/router';
+import { __experimentalHStack as HStack } from '@wordpress/components';
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../lock-unlock';
+import { __ } from '@wordpress/i18n';
+import type { BasePost } from '../../types';
+import type { DataViewRenderFieldProps } from '@wordpress/dataviews';
+import { getItemTitle } from '../../actions/utils';
+
+const { useHistory } = unlock( routerPrivateApis );
+
+const Media = ( {
+	id,
+	size = [ 'large', 'medium', 'thumbnail' ],
+	...props
+}: {
+	id: number;
+	size?: Array< 'large' | 'medium' | 'full' | 'thumbnail' >;
+} & Record< string, any > ) => {
+	const { record: media } = useEntityRecord< Attachment >(
+		'root',
+		'media',
+		id
+	);
+
+	const currentSize =
+		size.find( ( s ) => !! media?.media_details?.sizes[ s ] ) ?? '';
+
+	const mediaUrl =
+		media?.media_details?.sizes[ currentSize ]?.source_url ||
+		media?.source_url;
+
+	if ( ! mediaUrl ) {
+		return null;
+	}
+
+	return <img { ...props } src={ mediaUrl } alt={ media.alt_text } />;
+};
+
+const getComponentToDisplay = ( {
+	item,
+	url,
+	title,
+	onClick,
+}: {
+	item: BasePost;
+	title: string | undefined;
+	url: string | undefined;
+	onClick: () => void;
+} ) => ( {
+	grid: {
```

Comment:
How this component will be displayed when the view type is unknown? We aim to allow 3rd parties to register/unregister layouts for the DataViews.

Can we make it so this render function is independent from the view type? Thinking out loud: perhaps this render function always return an `img` and the layout can add wrappers based on the field type.